lib/ddtrace/contrib/sneakers/tracer.rb

```diff
@@ -0,0 +1,53 @@
+# frozen_string_literal: true
+
+require 'ddtrace/contrib/analytics'
+
+module Datadog
+  module Contrib
+    module Sneakers
+      # Tracer is a Seanerks server-side middleware which traces executed jobs
+      class Tracer
+        def initialize(app, *args)
+          @app = app
+          @args = args
+        end
+
+        def call(deserialized_msg, delivery_info, metadata, handler)
+          trace_options = {
+            service:   configuration[:service_name],
+            span_type: Datadog::Ext::AppTypes::WORKER
+          }
+          request_span = tracer.trace(Ext::SPAN_JOB, trace_options)
+
+          # Set analytics sample rate
+          if Datadog::Contrib::Analytics.enabled?(configuration[:analytics_enabled])
+            Datadog::Contrib::Analytics.set_sample_rate(request_span, configuration[:analytics_sample_rate])
+          end
+
+          # Measure service stats
+          Contrib::Analytics.set_measured(request_span)
+
+          request_span.set_tag(Ext::TAG_JOB_ROUTING_KEY, delivery_info.routing_key)
+          request_span.set_tag(Ext::TAG_JOB_QUEUE, delivery_info.consumer.queue.name)
+          request_span.set_tag(Ext::TAG_JOB_BODY, deserialized_msg)
```

Comment:
I'm guessing it's possible that `deserialized_msg` contains sensitive or identifiable information (e.g. email address, credit card number).
If that's the case, I suggest we don't record this tag by default, but add an option at integration level to enable its collection, similar to Sidekiq's `tag_args` option.

Reply:
Indeed good point.

I added an `tag_body` option and also was able to add the resource name to the span